qiskit_experiments/curve_analysis/curve_data.py

```diff
@@ -106,26 +100,22 @@ class FitData:
     # Y data range
     y_range: Tuple[float, float]
 
-    def fitval(self, key: str, unit: Optional[str] = None) -> FitVal:
+    def fitval(self, key: str) -> UFloat:
         """A helper method to get fit value object from parameter key name.
 
         Args:
             key: Name of parameters to extract.
-            unit: Optional. Unit of this value.
 
         Returns:
-            FitVal object.
+            UFloat object. This object can be operated as standard
+            Python float object with automatic error propagation.
 
         Raises:
             ValueError: When specified parameter is not defined.
         """
         try:
             index = self.popt_keys.index(key)
-            return FitVal(
-                value=self.popt[index],
-                stderr=self.popt_err[index],
-                unit=unit,
-            )
+            return self.popt[index]
```

Comment:
Yes this should be okey. Unit eventually appears in `AnalysisResultData`. We cannot give `UFloat` a unit information (we can store unit in `tag` field but it breaks string representation, e.g. it may look like <Hz = 123 +/- 45>)